src/openms/include/OpenMS/ANALYSIS/XLMS/OpenProXLUtils.h

```diff
@@ -0,0 +1,302 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2016.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Eugen Netz $
+// $Authors: Eugen Netz $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_ANALYSIS_XLMS_OPENPROXLUTILS
+#define OPENMS_ANALYSIS_XLMS_OPENPROXLUTILS
+
+#include <OpenMS/KERNEL/StandardTypes.h>
+#include <OpenMS/CHEMISTRY/ResidueModification.h>
+#include <OpenMS/FORMAT/FASTAFile.h>
+#include <OpenMS/CHEMISTRY/EnzymaticDigestion.h>
+#include <numeric>
+
+namespace OpenMS
+{
+
+  struct ProteinProteinCrossLink
```

Comment:
Here I would like to see documentation with doxygen Syntax. Otherwise I wont be able to tell easily what the difference between the structs here are `ProteinProteinCrossLink` and `CrossLinkSpectrumMatch` . These two structs seem to be mostly data storage containers, so I think using a `struct` and public members *could* be ok here, even though the OpenMS codeing style does not encourage this.